bgpd/bgp_route.c

```diff
@@ -5159,13 +5159,26 @@ void bgp_update(struct peer *peer, const struct prefix *p, uint32_t addpath_id,
 		}
 
 	new_attr = *attr;
+	if (CHECK_FLAG(peer->af_flags[afi][safi], PEER_FLAG_SOFT_RECONFIG)) {
```

Comment:
it's a little confusing that there are sort of three blocks of soft-reconfig logic in here - the one above, this new one, and the block below. Would it be clearer to do the evpn+soft-reconfig things in one block, in one place?

Reply:
There needs to be at least two blocks, due to interning shenanigans. See my reply to Donatas, the issue is much deeper than expected, and this does fix the issue, but I'm still trying to figure out why. (I did confirm we have a double-free on this struct when it is interned, I'm still trying to hunt the specific place).